src/GraphQL/Types/ISchema.cs

```diff
@@ -110,6 +110,25 @@ public interface ISchema : IProvideMetadata, IProvideDescription
         /// </summary>
         void RegisterType(Type type);
 
+        /// <summary>
+        /// Registers type mapping from CLR type to GraphType.
+        /// <br/>
+        /// These mappings are used for type inference when constructing fields using expressions:
+        /// <br/>
+        /// <c>
+        /// Field(x => x.Filters);
+        /// </c>
+        /// </summary>
+        /// <param name="clrType">The CLR property type from which to infer the GraphType.</param>
+        /// <param name="graphType">Inferred GraphType.</param>
+        void RegisterTypeMapping(Type clrType, Type graphType);
+
+        /// <summary>
+        /// Returns all registered by <see cref="RegisterTypeMapping"/> type mappings.
+        /// </summary>
+        /// <param name="includeBuiltInScalarMappings">Is it required to return built-in mappings for scalars.</param>
+        IEnumerable<(Type clrType, Type graphType)> EnumerateTypeMappings(bool includeBuiltInScalarMappings = false);
```

Comment:
Can we do better?  The other collections properties on the schema are: `Directives`, `AllTypes`, `AdditionalTypes`.  Why does this have to be a method, with an argument, prefixed with `Enumerate`?  How about simply:

```cs
IEnumerable<(Type clrType, Type graphType)> TypeMappings { get; }
```

It can return built in scalar mappings (unless they have been overridden).  Or we can choose to not include built-in mappings.  Or we can add another property:

```cs
IEnumerable<(Type clrType, Type graphType)> BuiltInTypeMappings { get; }
```

Reply:
done